contents/code/main.js

```diff
@@ -9,217 +9,208 @@
-}
-
-function _IsVerticallyMaximized() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.height === screenGeometry.height) {
-        return true
-    }
-    return false
-}
-
-function _IsHorizontallyMaximized() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.width === screenGeometry.width) {
-        return true
-    }
-    return false
-}
-
-function _IsMaximized() {
-    return _IsHorizontallyMaximized() && _IsVerticallyMaximized()
-}
-
-function _IsTiledToTop() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.height === (screenGeometry.height / 2) && clientGeometry.y === 0) {
-        return true
-    }
-    return false
-}
-
-function _IsTiledTop() {
-    return _IsTiledToTop() && _IsHorizontallyMaximized()
-}
-
-function _IsTiledToBottom() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.height === (screenGeometry.height / 2) && clientGeometry.y === (screenGeometry.height / 2)) {
-        return true
-    }
-    return false
-}
-
-function _IsTiledBottom() {
-    return _IsTiledToBottom() && _IsHorizontallyMaximized()
-}
-
-function _IsTiledToLeft() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.width === (screenGeometry.width / 2) && clientGeometry.x === 0) {
-        return true
-    }
-    return false
-}
-
-function _IsTiledLeft() {
-    return _IsTiledToLeft() && _IsVerticallyMaximized()
-}
-
-function _IsTiledToRight() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.width === (screenGeometry.width / 2) && clientGeometry.x === (screenGeometry.width / 2)) {
-        return true
-    }
-    return false
-}
-
-function _IsTiledRight() {
-    return _IsTiledToRight() && _IsVerticallyMaximized()
-}
-
-function _IsTiledToQuadrant() {
-    const screenGeometry = _GetScreenGeometry()
-    const clientGeometry = _GetClientGeometryOnScreen()
-    if (clientGeometry.width === (screenGeometry.width / 2) && clientGeometry.height === (screenGeometry.height/ 2)) {
-        return true
-    }
-    return false
-}
-
-function _IsTiledTopLeft() {
-    return _IsTiledToTop() && _IsTiledToLeft()
-}
-
-function _IsTiledTopRight() {
-    return _IsTiledToTop() && _IsTiledToRight()
-}
-
-function _IsTiledBottomLeft() {
-    return _IsTiledToBottom() && _IsTiledToLeft()
-}
-
-function _IsTiledBottomRight() {
-    return _IsTiledToBottom() && _IsTiledToRight()
-}
-
-var QuickTileUp = function() {
-    // L > TL
-    if (_IsTiledLeft()) {
-        workspace.slotWindowQuickTileTopLeft()
+function _tileEvaluator() {
```

Comment:
I introduced this object so we don't have to fetch the current state over and over for all the comparisons.